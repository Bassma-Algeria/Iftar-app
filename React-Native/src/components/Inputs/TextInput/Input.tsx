import React from 'react';
import {
  TextInput,
  View,
  KeyboardTypeOptions,
  Image,
  ImageSourcePropType,
} from 'react-native';

import {Shadow} from 'react-native-shadow-2';

import {styles} from '../Input.style';

interface Props {
  label: string;
  keyboardType?: KeyboardTypeOptions | undefined;
  icon: ImageSourcePropType;
}

const Input: React.FC<Props> = ({label, icon, keyboardType}) => {
  const [input, onInputChange] = React.useState<string>('');
  return (
    <View style={styles.inputHolder}>
      <Shadow distance={6} startColor={'#00000010'}>
        <View style={styles.inputContainer}>
          <Image source={icon} resizeMode="contain" />
          <TextInput
            style={styles.input}
            onChangeText={onInputChange}
            value={input}
            placeholder={label}
            keyboardType={keyboardType}
          />
        </View>
      </Shadow>
    </View>
  );
};

export {Input};
