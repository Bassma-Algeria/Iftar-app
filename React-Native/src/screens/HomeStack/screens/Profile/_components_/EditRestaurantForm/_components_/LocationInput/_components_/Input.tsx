import React from 'react';
import {TextInput, StyleProp, ViewStyle} from 'react-native';

import {styles} from '../../../../../Profile.style';

import {COLORS} from '../../../../../../../../../theme/Colors';

import {BaseInput} from '../../../../../../../../../components/Inputs/BaseInput';

interface Props {
  placeholder: string;
  value: string;
  style?: StyleProp<ViewStyle>;
  error?: string;
  disable?: boolean;
}

const Input: React.FC<Props> = props => {
  return (
    <BaseInput {...props} backgroundColor="beige">
      {props.children}

      <TextInput
        style={styles.locationInput}
        value={props.value}
        placeholder={props.placeholder}
        placeholderTextColor={COLORS.greyShade}
        editable={!props.disable}
      />
    </BaseInput>
  );
};

export {Input};
