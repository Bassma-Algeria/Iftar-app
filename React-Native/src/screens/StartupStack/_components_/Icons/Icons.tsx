import React from 'react';
import {Image} from 'react-native';
import {ICONS} from '../../../../utils/constants/Icons.ts';
interface Props {}
const Email: React.FC<Props> = () => {
  return <Image source={ICONS.email} resizeMode="contain" />;
};
const Phone: React.FC<Props> = () => {
  return <Image source={ICONS.phoneLightColor} resizeMode="contain" />;
};
const OpenEye: React.FC<Props> = () => {
  return <Image source={ICONS.openEye} resizeMode="contain" />;
};
const ClosedEye: React.FC<Props> = () => {
  return <Image source={ICONS.closedEye} resizeMode="contain" />;
};
export {Email, Phone, OpenEye, ClosedEye};
