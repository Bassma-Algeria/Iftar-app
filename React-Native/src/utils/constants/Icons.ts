const ICONS = {
  moon: require('../../assets/icons/moon.png'),
  restaurantMarker: require('../../assets/icons/restaurant-marker.png'),
  clock: require('../../assets/icons/clock.png'),
  clockBrown: require('../../assets/icons/clock-brown.png'),
  location: require('../../assets/icons/location.png'),
  phone: require('../../assets/icons/phone.png'),
  leftArrow: require('../../assets/icons/left-arrow.png'),
  email: require('../../assets/icons/email.png'),
  closedEye: require('../../assets/icons/closed-eye.png'),
  openEye: require('../../assets/icons/open-eye.png'),
  phoneLightColor: require('../../assets/icons/phone-light-color.png'),
  myLocation: require('../../assets/icons/my-location.png'),
  search: require('../../assets/icons/search.png'),
  profile: require('../../assets/icons/profile.png'),
  plus: require('../../assets/icons/plus.png'),
  locationFilled: require('../../assets/icons/location-filled.png'),
  profilePrimary: require('../../assets/icons/profile-primary.png'),
  edit: require('../../assets/icons/edit.png'),
  addPicture: require('../../assets/icons/add-picture.png'),
  done: require('../../assets/icons/done.png'),
  trash: require('../../assets/icons/trash.png'),
  rightArrow: require('../../assets/icons/right-arrow.png'),
  profileBrown: require('../../assets/icons/profile-brown.png'),
};

export {ICONS};
