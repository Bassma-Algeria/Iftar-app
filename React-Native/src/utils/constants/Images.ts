const IMAGES = {
  latterns1: require('../../assets/images/latterns-1.png'),
  latterns2: require('../../assets/images/latterns-2.png'),
};

export {IMAGES};
