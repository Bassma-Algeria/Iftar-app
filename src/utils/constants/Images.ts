const IMAGES = {
  latterns1: require('../../assets/images/latterns-1.png'),
  latterns2: require('../../assets/images/latterns-2.png'),
  noImage: require('../../assets/images/no-image.png'),
};

export {IMAGES};
