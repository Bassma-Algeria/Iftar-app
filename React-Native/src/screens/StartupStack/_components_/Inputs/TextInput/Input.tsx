import React from 'react';
import {TextInput, View} from 'react-native';
import {styles} from './Input.style';
interface Props {}
const Input: React.FC<Props> = ({label, children, keyboardType}) => {
  const [input, onInputChange] = React.useState<string>('');
  return (
    <>
      <View style={styles.inputContainer}>
        {children}
        <TextInput
          style={styles.input}
          onChangeText={onInputChange}
          value={input}
          placeholder={label}
          keyboardType={keyboardType}
        />
      </View>
    </>
  );
};

export {Input};
