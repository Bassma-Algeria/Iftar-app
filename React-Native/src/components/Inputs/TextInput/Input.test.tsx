import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';

import {ICONS} from '../../../utils/constants/Icons';

import {Input} from './Input';
import {TextInput} from 'react-native';

describe('Input component', () => {
  const props: any = {placeholder: 'Email', icon: ICONS.email};

  it('should render correctly, with the icon in the left by default', () => {
    const tree = renderer.create(<Input {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the icon position', () => {
    const tree = renderer.create(<Input {...props} iconPosition={'right'} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the default style', () => {
    const tree = renderer
      .create(<Input {...props} style={{margin: 40}} iconPosition={'right'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should show the error text when there is an error passed in the props', () => {
    const tree = renderer
      .create(<Input {...props} style={{margin: 40}} error={'someError'} iconPosition={'right'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the borderRadius of the input', () => {
    const tree = renderer.create(<Input {...props} radius={30} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call onTextChange each time the input value changed', () => {
    const onTextChange = jest.fn();

    const instance = render(
      <Input {...props} iconPosition={'right'} onTextChange={onTextChange} />,
    );
    const TargetInput = instance.UNSAFE_getByType(TextInput);

    fireEvent(TargetInput, 'changeText', 's');

    expect(onTextChange).toBeCalledTimes(1);
    expect(onTextChange).toBeCalledWith('s');

    fireEvent(TargetInput, 'changeText', 'sa');
    fireEvent(TargetInput, 'changeText', 'sari');
    fireEvent(TargetInput, 'changeText', 'sarili');

    expect(onTextChange).toBeCalledTimes(4);
    expect(onTextChange).toBeCalledWith('sarili');
  });
});
