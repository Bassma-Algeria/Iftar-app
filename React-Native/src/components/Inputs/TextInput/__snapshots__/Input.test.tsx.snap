// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
      "marginTop": 20,
      "width": "85%",
    }
  }
>
  <View
    pointerEvents="box-none"
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <RNSVGSvgView
        bbHeight="100%"
        bbWidth={7}
        focusable={false}
        height="100%"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": -6,
              "position": "absolute",
              "top": 10,
            },
            Object {
              "flex": 0,
              "height": "100%",
              "width": 7,
            },
          ]
        }
        width={7}
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="left"
              x1="1"
              x2="0"
              y1="0"
              y2="0"
            />
          </RNSVGDefs>
          <RNSVGRect
            fill={
              Array [
                1,
                "left",
              ]
            }
            height="100%"
            propList={
              Array [
                "fill",
              ]
            }
            width={6}
            x={0}
            y={-20}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        bbHeight="100%"
        bbWidth={7}
        focusable={false}
        height="100%"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": "100%",
              "position": "absolute",
              "top": 10,
            },
            Object {
              "flex": 0,
              "height": "100%",
              "width": 7,
            },
          ]
        }
        width={7}
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="right"
              x1="0"
              x2="1"
              y1="0"
              y2="0"
            />
          </RNSVGDefs>
          <RNSVGRect
            fill={
              Array [
                1,
                "right",
              ]
            }
            height="100%"
            propList={
              Array [
                "fill",
              ]
            }
            width={6}
            x={0}
            y={-20}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        bbHeight={7}
        bbWidth="100%"
        focusable={false}
        height={7}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": 10,
              "position": "absolute",
              "top": "100%",
            },
            Object {
              "flex": 0,
              "height": 7,
              "width": "100%",
            },
          ]
        }
        width="100%"
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="bottom"
              x1="0"
              x2="0"
              y1="0"
              y2="1"
            />
          </RNSVGDefs>
          <RNSVGRect
            fill={
              Array [
                1,
                "bottom",
              ]
            }
            height={6}
            propList={
              Array [
                "fill",
              ]
            }
            width="100%"
            x={-20}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        bbHeight={7}
        bbWidth="100%"
        focusable={false}
        height={7}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": 10,
              "position": "absolute",
              "top": -6,
            },
            Object {
              "flex": 0,
              "height": 7,
              "width": "100%",
            },
          ]
        }
        width="100%"
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="top"
              x1="0"
              x2="0"
              y1="1"
              y2="0"
            />
          </RNSVGDefs>
          <RNSVGRect
            fill={
              Array [
                1,
                "top",
              ]
            }
            height={6}
            propList={
              Array [
                "fill",
              ]
            }
            width="100%"
            x={-20}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        bbHeight={17}
        bbWidth={17}
        focusable={false}
        height={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": -6,
              "position": "absolute",
              "top": -6,
            },
            Object {
              "flex": 0,
              "height": 17,
              "width": 17,
            },
          ]
        }
        width={17}
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGRadialGradient
              cx={16}
              cy={16}
              fx={16}
              fy={16}
              gradient={
                Array [
                  0.625,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={1}
              name="topLeft"
              rx={16}
              ry={16}
            />
          </RNSVGDefs>
          <RNSVGPath
            d="M0,16 a16,16 0 0 1 16 -16 v6 a10,10 0 0 0 -10,10 h-6 Z"
            fill={
              Array [
                1,
                "topLeft",
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        bbHeight={17}
        bbWidth={17}
        focusable={false}
        height={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": "100%",
              "position": "absolute",
              "top": -6,
              "transform": Array [
                Object {
                  "translateX": -10,
                },
              ],
            },
            Object {
              "flex": 0,
              "height": 17,
              "width": 17,
            },
          ]
        }
        width={17}
      >
        <RNSVGGroup
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
        >
          <RNSVGDefs>
            <RNSVGRadialGradient
              cx={0}
              cy={16}
              fx={0}
              fy={16}
              gradient={
                Array [
                  0.625,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={1}
              name="topRight"
              rx={16}
              ry={16}
            />
          </RNSVGDefs>
          <RNSVGPath
            d="M0,0 a16,16 0 0 1 16,16 h-6 a10,10 0 0 0 -10,-10 v-6 Z"
            fill={
              Array [
                1,
                "topRight",
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        bbHeight={17}
        bbWidth={17}
        focusable={false}
        height={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": -6,
              "position": "absolute",
              "top": "100%",
              "transform": Array [
                Object {
                  "translateY": -10,
                },
              ],
            },
            Object {
              "flex": 0,
              "height": 17,
              "width": 17,
            },
          ]
        }
        width={17}
      >
        <RNSVGGroup
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
        >
          <RNSVGDefs>
            <RNSVGRadialGradient
              cx={16}
              cy={0}
              fx={16}
              fy={0}
              gradient={
                Array [
                  0.625,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={1}
              name="bottomLeft"
              rx={16}
              ry={16}
            />
          </RNSVGDefs>
          <RNSVGPath
            d="M16,16 a16,16 0 0 1 -16,-16 h6 a10,10 0 0 0 10,10 v6 Z"
            fill={
              Array [
                1,
                "bottomLeft",
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        bbHeight={17}
        bbWidth={17}
        focusable={false}
        height={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": "100%",
              "position": "absolute",
              "top": "100%",
              "transform": Array [
                Object {
                  "translateX": -10,
                },
                Object {
                  "translateY": -10,
                },
              ],
            },
            Object {
              "flex": 0,
              "height": 17,
              "width": 17,
            },
          ]
        }
        width={17}
      >
        <RNSVGGroup
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
        >
          <RNSVGDefs>
            <RNSVGRadialGradient
              cx={0}
              cy={0}
              fx={0}
              fy={0}
              gradient={
                Array [
                  0.625,
                  251658240,
                  1,
                  0,
                ]
              }
              gradientTransform={null}
              gradientUnits={1}
              name="bottomRight"
              rx={16}
              ry={16}
            />
          </RNSVGDefs>
          <RNSVGPath
            d="M16,0 a16,16 0 0 1 -16,16 v-6 a10,10 0 0 0 10,-10 h6 Z"
            fill={
              Array [
                1,
                "bottomRight",
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "width": "100%",
          }
        }
      >
        <Image
          resizeMode="contain"
        />
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "fontFamily": "Cairo-Regular",
              "height": 60,
              "textAlign": "right",
              "width": "80%",
            }
          }
          value=""
        />
      </View>
    </View>
  </View>
</View>
`;
